azurerm/resource_arm_bot_channel_slack.go

```diff
@@ -0,0 +1,219 @@
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/validate"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/features"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmBotChannelSlack() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmBotChannelSlackCreate,
+		Read:   resourceArmBotChannelSlackRead,
+		Delete: resourceArmBotChannelSlackDelete,
+		Update: resourceArmBotChannelSlackUpdate,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"resource_group_name": azure.SchemaResourceGroupName(),
+
+			"location": azure.SchemaLocation(),
+
+			"bot_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validate.NoEmptyStrings,
+			},
+
+			"client_id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validate.NoEmptyStrings,
+			},
+
+			"client_secret": {
+				Type:         schema.TypeString,
+				Required:     true,
+				Sensitive:    true,
+				ValidateFunc: validate.NoEmptyStrings,
+			},
+
+			"verification_token": {
+				Type:         schema.TypeString,
+				Required:     true,
+				Sensitive:    true,
+				ValidateFunc: validate.NoEmptyStrings,
+			},
+
+			"landing_page_url": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validate.NoEmptyStrings,
+			},
+		},
+	}
+}
+
+func resourceArmBotChannelSlackCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).bot.ChannelClient
+	ctx := meta.(*ArmClient).StopContext
+
+	resourceGroup := d.Get("resource_group_name").(string)
+	botName := d.Get("bot_name").(string)
+
+	if features.ShouldResourcesBeImported() && d.IsNewResource() {
+		existing, err := client.Get(ctx, resourceGroup, string(botservice.ChannelNameSlackChannel), botName)
+		if err != nil {
+			if !utils.ResponseWasNotFound(existing.Response) {
+				return fmt.Errorf("Error checking for presence of creating Bot Channel Slack (Resource Group %q / Bot %q): %+v", resourceGroup, botName, err)
+			}
+		}
+		if existing.ID != nil && *existing.ID != "" {
+			return tf.ImportAsExistsError("azurerm_bot_channel_slack", *existing.ID)
+		}
+	}
+
+	channel := botservice.BotChannel{
+		Properties: botservice.SlackChannel{
+			Properties: &botservice.SlackChannelProperties{
+				ClientID:                utils.String(d.Get("client_id").(string)),
+				ClientSecret:            utils.String(d.Get("client_secret").(string)),
+				VerificationToken:       utils.String(d.Get("verification_token").(string)),
+				LandingPageURL:          utils.String(d.Get("landing_page_url").(string)),
+				IsEnabled:               utils.Bool(true),
+				RegisterBeforeOAuthFlow: utils.Bool(true),
+			},
+			ChannelName: botservice.ChannelNameSlackChannel1,
+		},
+		Location: utils.String(d.Get("location").(string)),
```

Comment:
we need to normalize this via:

```suggestion
		Location: utils.String(azure.NormalizeLocation(d.Get("location").(string))),
```